### PR TITLE
Add missing method impl interface

### LINKED ISSUE #459: AddMissingMethodImplementation adds method implementations to Interfaces
## What version of OpenRewrite are you using?

I am using

- Gradle plugin v6.11.2
- rewrite-bom v2.9.0
- rewrite-migrate-java

## How are you running OpenRewrite?

I have created a minimal test case to reproduce the issue here: https://github.com/darylrobbins/rewrite-test/tree/missingimp

I am using the Gradle plugin, and my project is a single module project.
```gradle
plugins {
    id("java")
    id("org.openrewrite.rewrite") version "6.11.2"
}

group = "org.sample.rewrite"
version = "1.0-SNAPSHOT"

repositories {
    mavenCentral()
}

dependencies {
    testImplementation(platform("org.junit:junit-bom:5.10.0"))
    testImplementation("org.junit.jupiter:junit-jupiter")
    rewrite(platform("org.openrewrite.recipe:rewrite-recipe-bom:2.9.0"))
    rewrite("org.openrewrite.recipe:rewrite-migrate-java")
}

dependencies {

}

rewrite {
    activeRecipe(
        "org.sample.rewrite.AddMissingMethodImplementation"
    )
}

tasks.test {
    useJUnitPlatform()
}
```

## What is the smallest, simplest way to reproduce the problem?

Here is the recipe:
```yaml
type: specs.openrewrite.org/v1beta/recipe
name: org.sample.rewrite.AddMissingMethodImplementation
displayName: Adds missing JDBC interface methods.
description: Add method implementations stubs to classes that implement JDBC interfaces.
recipeList:
  - org.openrewrite.java.migrate.AddMissingMethodImplementation:
      fullyQualifiedClassName: javax.sql.CommonDataSource
      methodPattern: "*..* getParentLogger()"
      methodTemplateString: "public java.util.logging.Logger getParentLogger() { \n\t// TODO Auto-generated method stub\n return null; }"
```

Here is a Java interface in the codebase:
```java
public interface MyDatasourceInterface extends DataSource {
}
```

## What did you expect to see?

```java
public interface MyDatasourceInterface extends DataSource {
}

```

## What did you see instead?

```java
public interface MyDatasourceInterface extends DataSource {
    public java.util.logging.Logger getParentLogger() {
        // TODO Auto-generated method stub
        return null;
    }
}
```

### PR DESCRIPTION
**Pull request by darylrobbins**
## What's changed?

* AddMissingMethodImplementation will no longer add methods to interfaces
* Added tests for AddMissingMethodImplementation

## What's your motivation?

Fixes #459

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/test/java/org/openrewrite/java/migrate/AddMissingMethodImplementationTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import org.openrewrite.DocumentExample;
+import static org.openrewrite.java.Assertions.java;
+import static org.openrewrite.java.Assertions.javaVersion;
+
+class AddMissingMethodImplementationTest implements RewriteTest {
+import org.openrewrite.Issue;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+public class AddMissingMethodImplementationTest implements RewriteTest {
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/AddMissingMethodImplementationTest.java:27]:
```suggestion
import org.openrewrite.DocumentExample;
import org.openrewrite.Issue;
import org.openrewrite.test.RecipeSpec;
import org.openrewrite.test.RewriteTest;

import static org.openrewrite.java.Assertions.java;
import static org.openrewrite.java.Assertions.javaVersion;

class AddMissingMethodImplementationTest implements RewriteTest {
```

File: src/test/java/org/openrewrite/java/migrate/AddMissingMethodImplementationTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate;
+
+import static org.openrewrite.java.Assertions.java;
+import static org.openrewrite.java.Assertions.javaVersion;
+
```
[the remark this answers is not in the document]
Comment (darylrobbins) [reply src/test/java/org/openrewrite/java/migrate/AddMissingMethodImplementationTest.java:20]: Isn't the before/after the same?

**Comment by darylrobbins** [comment]
UpdateBeanManagerMethodsTest > fireEvent() is failing for both of the PR's so this doesn't look like anything to do with my changes. FWIW, the tests all pass for me locally.

**Review by timtebeek — APPROVED** [review]
Thanks a lot for adding the tests here in addition to fixing the underlying issue. It seems our automated PR review was thrown off a bit when correcting the imports; we expect to see the static imports added after any regular imports, whereas your IDE settings seem to put them first. No problem at all; we'll fix that on PRs as they come in. Glad to have you on board here!

**Pull request merged**